nemo/lightning/pytorch/trainer.py

```diff
@@ -7,9 +8,35 @@
 from nemo.lightning.fabric.conversion import to_fabric
 from nemo.lightning.fabric.fabric import Fabric
 from nemo.lightning.io.mixin import IOMixin, serialization, track_io
+from nemo.lightning.pytorch.callbacks import MegatronProgressPrinter
 
 
 class Trainer(pl.Trainer, IOMixin):
+    def __init__(
```

Comment:
Let's change this design. I really would prefer not to overwrite the `__init__` here since we loose auto-complete. 

Since it's megatron-specific, it can be moved inside the `MegatronStrategy`. It can be put inside [this](https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/pytorch/strategies.py#L241) method.